components/apimgt/org.wso2.carbon.apimgt.eventing.hub/src/main/java/org/wso2/carbon/apimgt/eventing/hub/EventHubEventPublisher.java

```diff
@@ -38,18 +38,20 @@ public void init() {
     }
 
     @Override
-    public void publish(EventPublisherEvent eventPublisherEvent) {
-        boolean tenantFlowStarted = false;
-        try {
-            PrivilegedCarbonContext.startTenantFlow();
-            PrivilegedCarbonContext.getThreadLocalCarbonContext()
-                    .setTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME, true);
-            tenantFlowStarted = true;
-            ServiceReferenceHolder.getInstance().getOutputEventAdapterService().publish(
-                    EventHubEventPublisherConstants.EVENT_HUB_NOTIFICATION_EVENT_PUBLISHER, null, eventPublisherEvent);
-        } finally {
-            if (tenantFlowStarted) {
-                PrivilegedCarbonContext.endTenantFlow();
+    public void publish(EventPublisherEvent eventPublisherEvent) throws EventPublisherException {
```

Comment:
We don't have to throw an exception here since none is thrown at the implementation level.
```
A general rule of implementing and extending is you can make your new class or interface "less restrictive" but not "more restrictive". If you think of the requirement to handle an exception as a restriction, an implementation that doesn't declare the exception is less restricive. Anybody who codes to the interface will not have trouble with your class. 
```